cpp/src/arrow/CMakeLists.txt

```diff
@@ -256,9 +256,9 @@ find_package(Backtrace)
 foreach(LIB_TARGET ${ARROW_LIBRARIES})
   target_compile_definitions(${LIB_TARGET}
     PRIVATE ARROW_EXPORTING)
-  if (Backtrace_FOUND)
+  if (Backtrace_FOUND AND NOT DEFINED ARROW_WITH_BACKTRACE)
     target_compile_definitions(${LIB_TARGET}
-      PRIVATE ARROW_WITH_BACKTRACE)
+      ARROW_WITH_BACKTRACE)
```

Comment:
Does this need to be transitive in public targets?

Reply:
Please use one of `INTERFACE|PUBLIC|PRIVATE`. This should only be transitive when we use it in headers (which we don't).

Reply:
can I propagate this up to a top-level option?

Reply:
theres no `execinfo.h`

Reply:
https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/logging.cc#L21